File: scripts/slot_popup.lua
--[[
--------------------------------------------------

This file is part of PLUCK.
You are free to use these files within your own resources.
Please retain the original credit and attached MIT license.
Support honest development.

Author: Case @ BOII Development
License: https://github.com/boiidevelopment/pluck/blob/main/LICENSE
GitHub: https://github.com/playingintraffic/pluck

--------------------------------------------------
]]

if core.is_server then

    local function slot_popup(source, data)
        if not data then return end
        TriggerClientEvent("pluck:cl:slot_popup", source, data)
    end

    exports("slot_popup", slot_popup)

end

if not core.is_server then

    local function slot_popup(data)
        if not data then return end
        SendNUIMessage({
            func = "slot_popup",
            payload = data
        })
    end

    exports("slot_popup", slot_popup)

    RegisterNetEvent("pluck:cl:slot_popup", function(data)
        slot_popup(data)
    end)

end

File: ui/js/components/slot_popup.js
/*
--------------------------------------------------

This file is part of PLUCK.
You are free to use these files within your own resources.
Please retain the original credit and attached MIT license.
Support honest development.

Author: Case @ BOII Development
License: https://github.com/boiidevelopment/pluck/blob/main/LICENSE
GitHub: https://github.com/playingintraffic/pluck

--------------------------------------------------
*/

import { resolve_image_path } from "/ui/js/utils.js";

export class SlotPopup {
    constructor(config = {}) {
        this.position = config.position || "top-center";
        this.container = null;
        this.init();
    }

    init() {
        if (this.container) return;

        this.container = document.createElement("div");
        this.container.className = `slot_popups slot_popups_${this.position}`;
        document.body.appendChild(this.container);
    }

    show({ item_id, image, quantity, action, rarity = "common" }) {
        const notification = document.createElement("div");
        notification.className = `slot_popup rarity_${rarity} slot_popup_${action}`;

        const action_symbol = action === "added" ? "+" : "-";
        const action_class = action === "added" ? "action_added" : "action_removed";

        notification.innerHTML = `
            <div class="slot_popup_inner">
                <div class="slot_popup_image">
                    <img src="${resolve_image_path(image, "/ui/assets/items/")}" />
                </div>
                <div class="slot_popup_quantity ${action_class}">
                    ${action_symbol}${Math.abs(quantity)}
                </div>
            </div>
        `;

        this.container.appendChild(notification);

        setTimeout(() => notification.classList.add("visible"), 10);

        setTimeout(() => {
            notification.classList.remove("visible");
            setTimeout(() => notification.remove(), 300);
        }, 2000);
    }

    destroy() {
        if (this.container) {
            this.container.remove();
            this.container = null;
        }
    }
}